packages/sdk/src/chains/EVM/utils/approvesAndChecksFns.ts

```diff
@@ -0,0 +1,86 @@
+  handlerAddress: string,
+): Promise<boolean> => {
+  try {
+    const approvedAddress = await tokenInstance.getApproved(tokenId);
+    const isApproved = approvedAddress === handlerAddress;
+    return isApproved;
+  } catch (error) {
+    console.log('Error on checkCurrentAllowanceOfErc20', error);
+    return Promise.reject(error);
+  }
+};
+
+/**
+ * Retrieves the current allowance of an ERC20 token for the specified sender and handler addresses.
+ *
+ * @example
+ * // Assuming you have a valid sender address, an ERC20 token instance (erc20Instance), and a handler address (erc20HandlerAddress)
+ * const currentAllowance = await checkCurrentAllowanceOfErc20(senderAddress, erc20Instance, erc20HandlerAddress);
+ * console.log('Current allowance:', currentAllowance);
+ *
+ * @param {string} senderAddress - The address of the token sender.
+ * @param {ERC20} erc20Instance - The ERC20 token instance used to query the allowance.
+ * @param {string} erc20HandlerAddress - The handler address for which the token allowance is checked.
+ * @returns {Promise<number>} A promise that resolves to a number representing the current allowance of the ERC20 token.
+ */
+export const checkCurrentAllowanceOfErc20 = async (
+  senderAddress: string,
+  erc20Instance: ERC20,
+  erc20HandlerAddress: string,
+): Promise<number> => {
+  try {
+    const currentAllowance = await erc20Instance.allowance(senderAddress, erc20HandlerAddress);
+
+    return Number(utils.formatUnits(currentAllowance, 18));
+  } catch (error) {
+    console.log('Error on checkCurrentAllowanceOfErc20', error);
+    return Promise.reject(error);
+  }
+};
+
+/**
+ * Approves the specified token instance for a given amount or tokenId and the handler address.
+ *
+ * @param {BigNumber} amountOrIdForApproval - The amount or tokenId to be approved.
+ * @param {ERC20 | ERC721MinterBurnerPauser} tokenInstance - The ERC20 or ERC721 token instance to be approved.
+ * @param {string} handlerAddress - The handler address for which the token is being approved.
+ * @param {BigNumber} gasPrice - The gas price for the approval transaction.
+ * @param {number} confirmations - The number of confirmations required before the transaction is considered successful.
+ * @param {ethers.PayableOverrides} overrides - Optional overrides for the transaction, such as gas price, gas limit,
+ * @returns {Promise<ContractReceipt>} A promise that resolves to a contract receipt once the approval transaction is executed.
+ */
+export const approve = async (
+  amountOrIdForApproval: BigNumber,
+  tokenInstance: ERC20 | ERC721MinterBurnerPauser,
+  handlerAddress: string,
+  confirmations: number,
+  overrides?: ethers.PayableOverrides,
+): Promise<ContractReceipt> => {
+  try {
+    const tx = await tokenInstance.approve(handlerAddress, amountOrIdForApproval, overrides);
+    const approvalAction = await tx.wait(confirmations);
+    return approvalAction;
+  } catch (error) {
+    console.log('Error on approve', error);
```

Comment:
Indeed. At the next line you can see `return Promise.reject(error);`
So here in case of error, promise is rejected and return an error, so an error could be handled by calling function

Reply:
but sitll, below we have a console.error, I guess for consistency we should change this one to the same

Reply:
Matijas comment is been the issue of not propagating errors, usage of `console` can be useful for debugging in production